### PR TITLE
Updated/Improved service client logging

### PR DESCRIPTION
**Pull request by tmahmood-microsoft**
In a typical service client API, the logging of the enter/exit doesn't match much with the log statements within.
For instance:

Logging.Enter(this, $**"Creating device: {device?.Id}"**, nameof(CreateAsync));
Logging.Error(this, $**"{nameof(CreateAsync)}** threw an exception: {ex}", nameof(CreateAsync));
Logging.Exit(this, $**"Creating device: {device?.Id}"**, nameof(CreateAsync));

### CODE REVIEW

File: iothub/service/src/Configurations/ConfigurationsClient.cs [hunk 1]
```diff
@@ -130,7 +130,7 @@ public virtual async Task<Configuration> GetAsync(string configurationId, Cancel
             catch (Exception ex)
             {
                 if (Logging.IsEnabled)
-                    Logging.Error(this, $"Getting configuration threw an exception: {ex}", nameof(GetAsync));
+                    Logging.Error(this, $"Getting configuration: {configurationId} threw an exception: {ex}", nameof(GetAsync));
```
Comment (drwill-ms) [inline iothub/service/src/Configurations/ConfigurationsClient.cs:133]: In this case you can remove the colon before the configuration Id as the sentence continues.

```suggestion
                    Logging.Error(this, $"Getting configuration {configurationId} threw an exception: {ex}", nameof(GetAsync));
```

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/src/Amqp/AmqpConnectionHandler.cs [hunk 2]
```diff
@@ -318,14 +318,14 @@ private async Task<ClientWebSocket> CreateClientWebSocketAsync(Uri websocketUri,
                         // Configure proxy server
                         websocket.Options.Proxy = webProxy;
                         if (Logging.IsEnabled)
-                            Logging.Info(this, $"{nameof(CreateClientWebSocketAsync)} Setting ClientWebSocket.Options.Proxy");
+                            Logging.Info(this, " Setting ClientWebSocket.Options.Proxy", nameof(CreateClientWebSocketAsync));
```
Comment (patilsnr) [inline iothub/service/src/Amqp/AmqpConnectionHandler.cs:321]: nit: is there an extra space here?

**Review by patilsnr — APPROVED** [review]
Looks good to me, I'd recommend getting sign off from someone else as well.

File: iothub/service/src/Amqp/AmqpConnectionHandler.cs (hunk 2 above)
Comment (tmahmood-microsoft) [reply iothub/service/src/Amqp/AmqpConnectionHandler.cs]: Fixed it. Thanks

**Pull request merged**